crates/core/tedge_agent/src/entity_manager/tests.rs

```diff
@@ -0,0 +1,628 @@
+use crate::entity_manager::server::EntityStoreResponse;
+use proptest::proptest;
+use std::collections::HashSet;
+use tedge_actors::Server;
+use tedge_api::entity::EntityMetadata;
+
+#[tokio::test]
+async fn new_entity_store() {
+    let (mut entity_store, _mqtt_output) = entity::server("device-under-test");
+
+    assert_eq!(
+        entity::get(&mut entity_store, "device/main//").await,
+        Some(EntityMetadata::main_device())
+    )
+}
+
+proptest! {
+    //#![proptest_config(proptest::prelude::ProptestConfig::with_cases(1000))]
+    #[test]
+    fn it_works_for_any_registration_order(registrations in model::walk(10)) {
+        tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async {
+            let (mut entity_store, _mqtt_output) = entity::server("device-under-test");
+            let mut state = model::State::new();
```

Comment:
Just FYI: Came across this [proptest-state-machine](https://proptest-rs.github.io/proptest/proptest/state-machine.html) crate which looked to be made for this kind of stuff.

TBH, after looking at some of their examples, I found this verbose version easier to comprehend than the abstractions and macro magic provided by that crate.